Dockerfile

```diff
@@ -1,4 +1,88 @@
-FROM scratch AS production
+FROM busybox AS sct
+
+WORKDIR /
+
 COPY ./scylla-bench .
-ENTRYPOINT ["/scylla-bench"]
-LABEL com.scylladb.loader-type=scylla-bench
+
+ENV PATH="/usr/local/bin:/:${PATH}"
+
+EXPOSE 6060
+EXPOSE 2121
+
+FROM golang:1.23-bookworm AS build
+
+ENV GO111MODULE=on
+ENV GOAMD64=v3
+ENV GOARM64=v8.3
+ENV CFLAGS="-O3"
+ENV CXXFLAGS="-O3"
+ENV DEBIAN_FRONTEND="noninteractive"
+ENV TZ="UTC"
+
+WORKDIR /scylla-bench
+
+COPY . .
+
+RUN apt-get update \
+    && apt-get upgrade -y  \
+    && apt-get install -y build-essential ca-certificates libc-dev \
+    && make build
+
+FROM build AS debug
+
+ENV GODEBUG="default=go1.23,cgocheck=1,disablethp=0,panicnil=0,http2client=1,http2server=1,asynctimerchan=0,madvdontneed=0"
+ENV PATH="/scylla-bench/bin:${PATH}"
+
+RUN apt-get install -y gdb gcc iputils-ping mlocate vim \
+    && make debug-build \
+    && go install github.com/go-delve/delve/cmd/dlv@latest \
+    && updatedb
+
+EXPOSE 6060
+EXPOSE 2121
+EXPOSE 2345
+
+ENTRYPOINT [ \
+    "dlv", "exec", "--log", "--listen=0.0.0.0:2345", "--allow-non-terminal-interactive", \
```

Comment:
what is this dlv ? where did it came for, and for what usage ?

Reply:
Sometimes there are bugs that can be caught only in docker, so having a debugger in it that can be used with external tools live vscode or goland is useful